### ISSUE TITLE
Duplicate sysname causes VLAN info to disappear 

### ISSUE THREAD

**Issue by Linwood-F** [netdisco/netdisco#418]
We had two switches, a DLink that was to be replaced with a NetGear.  We set up the new Netgear at a new IP, but the same sysname (snmp system name, it had no rDNS name yet) as the DLink.

Discovered the new one -- the Native VLAN column was filled in, but the VLAN column where trunked VLAN info should appear was blank on the new switch. 

## Expected Behavior

I expected the ports to show the same basic info, since the new switch had been modeled after the old; it did, mostly, but the trunked vlan column was blank on the new.

## Current Behavior
Blanked VLAN's. 

## Possible Solution
Upon changing the name on the second device (the device name inside the device), and re-discovering (I deleted first just to be clean), the VLAN's appeared.

## Steps to Reproduce (for bugs)
It should be as simple as having two switches with trunked VLAN's at the same name, though I did not have a good environment in which to experiment beyond this being-replaced switch.  If it is not obvious from the description, I can set up a test environment in a bit to try to replicate.

## Context
I doubt I have ever had this situation before; it sounded like something confusing internal device identifiers?   Each device in this case had only one IP and it was unique to each.

## Your Environment
App::Netdisco	2.39.30
SNMP::Info	3.61
DB Schema	53
PostgreSQL	9.5.12
Perl	5.22.1
ubuntu 16.06.4

**Comment by ollyg** [comment]
Hi @Linwood-F, thanks for the report. I am really surprised by this as we do not use the sysName anywhere in Netdisco as an identifier precisely because it is not unique (or can be blank!).

Are you able to set up a lab to reproduce this? I would be interested in what is stored in the DB device table row for each device, and sample device_port_vlan rows for each device.

Could it also be the case that the discover did not fully complete for some reason on the new device?

**Comment by Linwood-F** [comment]
I didn't explore much as frankly expected it to be a "well don't do that" type thing.

Let me see what I can set up.

**Comment by Linwood-F** [comment]
I've reproduced this in production in a similar scenario to when it first happened.

I have two devices, an old DLink at 192.168.1.202 and a newer Netgear at 192.168.1.203.

I deleted both from Netdisco.

I changed the sysname of the Netgear to be the same as the DLink.  I did not change the DNS/rDNS.

I did a discovery by IP address.

The port descriptions on the Netgear do not appear, nor the VLAN membership (though the native VLAN does appear).  The port name is a separate issue, that I think is a bug probably in SNMP.  I can't get them to appear at all.  The VLAN however is a different issue.

The "device" table for the two entries mostly looks correct, the right description, location, port count, model and brand for example, and the creation time looks right.

The "device_port" table for the two IP's looks more or less correct.

However, the device_vlan is not present at all for the netgear (i.e. the ip column does not have any values of 192.168.1.203), so naturally the device_port_vlan also has nothing for it.

I went into the netgear and changed it back to the correct (i.e. non-duplicated) name.

I redid the discovery (just hit the button) on the netgear; I did not delete it first.

As soon as I did that, the device_vlan and device_port_vlan tables were populated correctly for the netgear.

So there's definitely an issue that is caused by (only) the duplicated sysname field being duplicated.

I am not saying this is specific to the netgear or dlink, I just duplicated it in the scenario that had caused it before.  I'll try to duplicate with something else also.
